flight/PiOS/Common/pios_hal.c

```diff
@@ -1197,3 +1229,138 @@ void PIOS_HAL_ConfigureSerialSpeed(uintptr_t com_id,
 			break;
 	}
 }
+
+#ifdef PIOS_INCLUDE_I2C
+static int PIOS_HAL_ConfigureI2C(uint32_t *id,
+		const struct pios_i2c_adapter_cfg *cfg) {
+	if (!*id) {
+		// Not already initialized.
+		if (PIOS_I2C_Init(id, cfg)) {
+			PIOS_DEBUG_Assert(0);
+			AlarmsSet(SYSTEMALARMS_ALARM_I2C, SYSTEMALARMS_ALARM_CRITICAL);
+
+			return -1;
+		}
+	}
+
+	if (PIOS_I2C_CheckClear(*id) != 0) {
+		AlarmsSet(SYSTEMALARMS_ALARM_I2C, SYSTEMALARMS_ALARM_CRITICAL);
+		return -2;
+	}
+
+	if (AlarmsGet(SYSTEMALARMS_ALARM_I2C) ==
+			SYSTEMALARMS_ALARM_UNINITIALISED) {
+		AlarmsSet(SYSTEMALARMS_ALARM_I2C, SYSTEMALARMS_ALARM_OK);
+	}
+
+	PIOS_WDG_Clear();
+
+	return 0;
+}
+#endif // PIOS_INCLUDE_I2C
+
+int PIOS_HAL_ConfigureExternalBaro(HwSharedExtBaroOptions baro,
+		uint32_t *i2c_id,
+		const struct pios_i2c_adapter_cfg *i2c_cfg)
+{
+#if !defined(PIOS_INCLUDE_I2C)
+	return -1;
+#else
+	if (baro == HWSHARED_EXTBARO_NONE) {
+		return 0;
+	}
+
+	int ret = PIOS_HAL_ConfigureI2C(i2c_id, i2c_cfg);
+
+	if (ret) goto done;
+
+	switch(baro) {
+#if defined(PIOS_INCLUDE_BMP280)
+	case HWSHARED_EXTBARO_BMP280:
+		ret = PIOS_BMP280_Init(&external_bmp280_cfg, *i2c_id) ;
+
+		if (ret) goto done;
+
+		ret = PIOS_BMP280_Test();
+
+		if (ret) goto done;
+#endif // PIOS_INCLUDE_BMP280
+
+#if defined(PIOS_INCLUDE_MS5611)
+	case HWSHARED_EXTBARO_MS5611:
+		ret = PIOS_MS5611_Init(&external_ms5611_cfg, *i2c_id);
+		if (ret) goto done;
+
+		ret = PIOS_MS5611_Test();
+
+		if (ret) goto done;
+#endif // PIOS_INCLUDE_MS5611
+
+	default:
+		PIOS_Assert(0);	// Should be unreachable
```

Comment:
Do you think it's better to remove the default case and pick it up at compile time (holds as long as you don't pass in some arbitrary int rather than a member of the relevant enum), or wait until it fails at runtime if it is configured (solid against any input)?